src/image/prepare_single_image_build_matrix.py

```diff
@@ -103,13 +106,23 @@ def validate_image_trigger(data: dict) -> None:
 
         # set an output as a marker for later knowing if we need to release
         if "release" in builds[img_number]:
-            release_to = "true"
-            # the workflow GH matrix has a problem parsing nested JSON dicts
-            # so let's remove this field since we don't need it for the builds
-            builds[img_number]["release"] = "true"
+            min_eol = datetime.strptime(min(
+                v["end-of-life"] for v in builds[img_number]["release"].values()
+            ), "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc)
+            if min_eol < datetime.now(timezone.utc):
+                print("Track skipped because it reached its end of life")
+                del builds[img_number]
+                continue
+            else:
+                release_to = "true"
+                # the workflow GH matrix has a problem parsing nested JSON dicts
+                # so let's remove this field since we don't need it for the builds
+                builds[img_number]["release"] = "true"
```

Comment:
why not use the release_to variable?

Reply:
Using `release_to` variable will exempt the track in question from being built, but it will stop triggering vulnerability scans or other tests.